### PR TITLE
Misc/app configure timeout ceil threshold

### PR DESCRIPTION
**Pull request by hlecnt**
## What do these changes do?

The purpose of this PR is to restore the library performance when using low timeout values. See [Discussion 6008](https://github.com/aio-libs/aiohttp/discussions/6008).

## Are there changes in behavior for the user?

By default, the current library behavior is not changed. The user will have to enable the ceiling of low timeout values (values below 5 seconds).

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by asvetlov** [comment]
I like the idea.
Putting ceiling threshold into `ClientTimeout` object looks pretty elegant!

File: aiohttp/web_protocol.py [hunk 1]
```diff
@@ -144,6 +144,10 @@ class RequestHandler(BaseProtocol):
 
     max_headers -- Optional maximum header size
 
+    timeout_ceil_threshold -- Optional threshold value
+                              for to trigger ceiling of
```
Comment (Dreamsorcerer) [inline aiohttp/web_protocol.py:148]:
```suggestion
                              to trigger ceiling of
```

File: aiohttp/web_protocol.py [hunk 2]
```diff
@@ -144,6 +144,10 @@ class RequestHandler(BaseProtocol):
 
     max_headers -- Optional maximum header size
 
+    timeout_ceil_threshold -- Optional threshold value
```
Comment (Dreamsorcerer) [inline aiohttp/web_protocol.py]: `Optional value to specify threshold to ceil() timeout values`

Maybe also a link to documentation about the purpose of ceiling.

Should also update the note in the documentation to mention that the value is configurable.

**Comment by Dreamsorcerer** [comment]
Note that you still have this PR marked as a draft. Please update if ready for review.

File: aiohttp/web_protocol.py (hunk 2 above)
Comment (hlecnt) [reply aiohttp/web_protocol.py]: Done in the last commit

**Comment by hlecnt** [comment]
> Note that you still have this PR marked as a draft. Please update if ready for review.

The PR is ready to review now. I added some tests and a fix candidate in case of bad `handler_args` name (not existing in `RequestHandler` ctor), so that we keep compatibilty with older version because I'd really like to backport this PR in aiohttp `3.7` and `3.8`. Let me know.

Another thing, this CI [job](https://github.com/aio-libs/aiohttp/runs/4256200763?check_suite_focus=true) failed while `pre-commit` was OK.

**Comment by Dreamsorcerer** [comment]
> I'd really like to backport this PR in aiohttp `3.7` and `3.8`. Let me know.

As this is a new feature, I think this should go to 3.9. 3.8 should be bug fixes only.

**Review by asvetlov — APPROVED** [review]
LGTM

**Comment by asvetlov** [comment]
Backport fails, as expected.

@hlecnt if you want to see your change published soon, please feel free to make a manual backporting to 3.9 branch.
Otherwise, we should wait for 4.0 release, which is not ready yet and I have no estimation, sorry.

**Comment by hlecnt** [comment]
> Backport fails, as expected.
> 
> @hlecnt if you want to see your change published soon, please feel free to make a manual backporting to 3.9 branch. Otherwise, we should wait for 4.0 release, which is not ready yet and I have no estimation, sorry.

@asvetlov the PR for the backport to 3.9 is available [here](https://github.com/aio-libs/aiohttp/pull/6397).

Do you have an estimation date for the first 3.9 release ?

**Comment by asvetlov** [comment]
I think it should fly this year. We have 18 items in CHANGELOG.
Also, I want to publish 3.8.2 bugfix release. 3.8 is the latest version with Python 3.6 support.

Before making a release, I'd like to spend some time fixing #6227 and  #5621

**Pull request merged**